datastore/sqlite_datastore.nim

```diff
@@ -33,53 +39,149 @@ type
     containsStmt: ContainsStmt
     deleteStmt: DeleteStmt
     env: SQLite
+    getDataCol: BoundDataCol
     getStmt: GetStmt
     putStmt: PutStmt
     readOnly: bool
 
 const
-  IdType = "TEXT"
-  DataType = "BLOB"
-  TimestampType = "INTEGER"
-
   dbExt* = ".sqlite3"
-  tableTitle* = "Store"
+  tableName* = "Store"
+
+  idColName = "id"
+  dataColName = "data"
+  timestampColName = "timestamp"
+
+  idColIndex = 0
+  dataColIndex = 1
+  timestampColIndex = 2
+
+  idColType = "TEXT"
+  dataColType = "BLOB"
+  timestampColType = "INTEGER"
 
   # https://stackoverflow.com/a/9756276
   # EXISTS returns a boolean value represented by an integer:
   # https://sqlite.org/datatype3.html#boolean_datatype
   # https://sqlite.org/lang_expr.html#the_exists_operator
   containsStmtStr = """
     SELECT EXISTS(
-      SELECT 1 FROM """ & tableTitle & """
-      WHERE id = ?
+      SELECT 1 FROM """ & tableName & """
+      WHERE """ & idColName & """ = ?
     );
   """
 
   createStmtStr = """
-    CREATE TABLE IF NOT EXISTS """ & tableTitle & """ (
-      id """ & IdType & """ NOT NULL PRIMARY KEY,
-      data """ & DataType & """,
-      timestamp """ & TimestampType & """ NOT NULL
+    CREATE TABLE IF NOT EXISTS """ & tableName & """ (
+      """ & idColName & """ """ & idColType & """ NOT NULL PRIMARY KEY,
+      """ & dataColName & """ """ & dataColType & """,
+      """ & timestampColName & """ """ & timestampColType & """ NOT NULL
     ) WITHOUT ROWID;
   """
 
   deleteStmtStr = """
-    DELETE FROM """ & tableTitle & """
-    WHERE id = ?;
+    DELETE FROM """ & tableName & """
+    WHERE """ & idColName & """ = ?;
   """
 
   getStmtStr = """
-    SELECT data FROM """ & tableTitle & """
-    WHERE id = ?;
+    SELECT """ & dataColName & """ FROM """ & tableName & """
+    WHERE """ & idColName & """ = ?;
   """
 
   putStmtStr = """
-    REPLACE INTO """ & tableTitle & """ (
-      id, data, timestamp
+    REPLACE INTO """ & tableName & """ (
+      """ & idColName & """,
+      """ & dataColName & """,
+      """ & timestampColName & """
     ) VALUES (?, ?, ?);
   """
 
+template checkColMetadata(s: RawStmtPtr, i: int, expectedName: string) =
+  let
+    colName = sqlite3_column_origin_name(s, i.cint)
+
+  if colName.isNil:
+    raise (ref Defect)(msg: "no column exists for index " & $i)
+
+  if $colName != expectedName:
+    raise (ref Defect)(msg: "original column name for index " & $i & " was \"" &
+      $colName & "\" but expected \"" & expectedName & "\"")
+
+proc idCol*(
+  s: RawStmtPtr,
+  index = idColIndex): BoundIdCol =
```

Comment:
I propose to drop default value in all 3 procs. It's useful only for `SELECT *` statements

Reply:
Changed.